src/engraving/rendering/score/systemheaderlayout.cpp

```
@@ -1377,44 +1399,55 @@ void SystemHeaderLayout::setInstrumentNames(System* system, LayoutContext& ctx)
 }
 
 void SystemHeaderLayout::setSharedPartNames(SharedPart* sharedPart, staff_idx_t startStaffIdx, System* system, LayoutContext& ctx,
-                                            bool longName, const Fraction& tick)
+                                            const Fraction& tick)
 {
-    InstrumentNameType type = longName ? InstrumentNameType::LONG : InstrumentNameType::SHORT;
+    InstrumentLabelVisibility visibility
+        = resolveInstrumentLabelVisibility(ctx.dom().staff(startStaffIdx), tick, ctx, ctx.state().firstSystem());
+    InstrumentNameType type = resolveInstrumentNameType(visibility);
+    bool hide = visibility == InstrumentLabelVisibility::HIDE;
+    bool longName = type == InstrumentNameType::LONG;
```

Cppcheck (2.21.0)

[error] There is an unknown macro here somewhere. Configuration is required. If Q_DECLARE_METATYPE is a macro then please configure it.

(unknownMacro)